primitives/utility/src/lib.rs

```diff
@@ -20,34 +20,74 @@
 #![cfg_attr(not(feature = "std"), no_std)]
 
 use codec::Encode;
-use cumulus_primitives_core::UpwardMessageSender;
-use sp_std::marker::PhantomData;
+use cumulus_primitives_core::{MessageSendError, UpwardMessageSender};
+use frame_support::traits::Get;
+use sp_std::{marker::PhantomData, prelude::*};
 use xcm::{latest::prelude::*, WrapVersion};
 
+pub trait PriceForParentDelivery {
+	fn price_for_parent_delivery(message: &Xcm<()>) -> MultiAssets;
+}
+
+impl PriceForParentDelivery for () {
+	fn price_for_parent_delivery(_: &Xcm<()>) -> MultiAssets {
+	    MultiAssets::new()
+	}
+}
+
+pub struct ConstantPrice<T>(PhantomData<T>);
```

Comment:
I'd like to upstream this struct to polkadot, but I've no idea where I should put it. I've thought of putting it in the `xcm` or `xcm-builder` crate, but it doesn't quite make sense since this is configuring something related to the message passing protocol, rather than XCM itself. I've also thought of putting it in parachain primitives, but am not quite sure whether this counts as a "primitive", and I don't know if the maintainers of it has a special process around introducing types there.

If we don't do anything and leave it as-is, we'd have multiple `ConstantPrice` structs lying around, which doesn't sound ideal.

Reply:
It's already in Polkadot. See `xcm_sender.rs`.